orchagent/orch.h

```diff
@@ -68,9 +68,10 @@ class Orch;
 class Executor : public Selectable
 {
 public:
-    Executor(Selectable *selectable, Orch *orch)
+    Executor(Selectable *selectable, Orch *orch, const string &name)
         : m_selectable(selectable)
         , m_orch(orch)
+        , m_name(name)
```

Comment:
>m_name [](start = 10, length = 6)

I believe the new name field is not necessary, and you could always get it from m_consumerMap key. Could you help explorer the possibility?

Reply:
Isn't it a good practice to give executor a fixed unique name? For now it is mainly for debugging, later it may also be used for state validation log.  
I noticed that there are even cases where empty string given to to map as key in current code,  all those should be changed.

Reply:
I agree on "even cases where empty string given to to map as key in current code, all those should be changed." and I will fix them.

Reply:
Fixed by [https://github.com/Azure/sonic-swss/pull/575](https://github.com/Azure/sonic-swss/pull/575)

---
In reply to: [209784061](https://github.com/Azure/sonic-swss/pull/554#discussion_r209784061) [](ancestors = 209784061)

Reply:
I didn't say there was bug there, at least it is working. But it is a bad design putting empty string as key.
The purpose of name related change in this PR is to facilitate debugging and logging via a meaningful name for each excecutor including those timers and Notifiers.  I found it especially useful during the warm restart testing since very often you need to trace specific executor whenever problem happened.